numba/np/arraymath.py

```diff
@@ -4040,13 +4041,24 @@ def histogram_impl(a, bins=10, range=None):
 
 iinfo = namedtuple('iinfo', _iinfo_supported)
 
+# Suppress deprecation warnings while keeping support MachAr
+with warnings.catch_warnings(record=True):
+    warnings.simplefilter("ignore", DeprecationWarning)
+    np_MachAr = np.MachAr
 
-@overload(np.MachAr)
+
+@overload(np_MachAr)
 def MachAr_impl():
-    f = np.MachAr()
+    f = np_MachAr()
+
     _mach_ar_data = tuple([getattr(f, x) for x in _mach_ar_supported])
 
+    np_122 = np.version.version >= '1.22'
+
     def impl():
+        if np_122:
+            raise NumbaDeprecationWarning(
+                "`np.MachAr` is deprecated (NumPy 1.22)")
```

Comment:
This warning should not be raised with `raise`.  Use `warnings.warn` instead. Otherwise, this will be raised like an exception, which is not how numpy behaves.

See example in https://github.com/numba/numba/blob/main/numba/core/dataflow.py#L55-L56